ElevenLabs/Packages/com.rest.elevenlabs/Runtime/TextToSpeech/TextToSpeechEndpoint.cs

```diff
@@ -256,6 +257,323 @@ void StreamCallback(Response partialResponse)
+        /// Note: Must be PCM format to stream audio in Unity!<br/>
+        /// Defaults to <see cref="OutputFormat.PCM_24000"/>.
+        /// </param>
+        /// <param name="optimizeStreamingLatency">
+        /// Optional, You can turn on latency optimizations at some cost of quality.
+        /// The best possible final latency varies by model.<br/>
+        /// Possible values:<br/>
+        /// 0 - default mode (no latency optimizations)<br/>
+        /// 1 - normal latency optimizations (about 50% of possible latency improvement of option 3)<br/>
+        /// 2 - strong latency optimizations (about 75% of possible latency improvement of option 3)<br/>
+        /// 3 - max latency optimizations<br/>
+        /// 4 - max latency optimizations, but also with text normalizer turned off for even more latency savings
+        /// (best latency, but can mispronounce eg numbers and dates).
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional, <see cref="CancellationToken"/>.
+        /// </param>
+        /// <returns>Downloaded clip path, and the loaded audio clip.</returns>
+        public async Task<(VoiceClip voiceClip, TimestampedTranscriptCharacter[] timestampedTranscriptCharacters)> StreamTextToSpeechWithTimestampsAsync(string text, Voice voice, Action<(AudioClip audioClip, TimestampedTranscriptCharacter[] timestampedTranscriptCharacters)> partialClipCallback, VoiceSettings voiceSettings = null, Model model = null, OutputFormat outputFormat = OutputFormat.PCM_24000, int? optimizeStreamingLatency = null, CancellationToken cancellationToken = default)
+            => await StreamTextToSpeechWithTimestampsAsync(new TextToSpeechRequest(voice, text, Encoding.UTF8, voiceSettings ?? voice.Settings ?? await client.VoicesEndpoint.GetDefaultVoiceSettingsAsync(cancellationToken), outputFormat, optimizeStreamingLatency, model), partialClipCallback, cancellationToken);
+
+        /// <summary>
+        /// Converts text into speech using a voice of your choice and returns audio as an audio stream.
+        /// </summary>
+        /// <param name="request"><see cref="TextToSpeechRequest"/>.</param>
+        /// <param name="partialClipCallback">
+        /// Optional, Callback to enable streaming audio as it comes in.<br/>
+        /// Returns partial <see cref="VoiceClip"/>.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional, <see cref="CancellationToken"/>.
+        /// </param>
+        /// <returns>Downloaded clip path, and the loaded audio clip.</returns>
+        public async Task<(VoiceClip voiceClip, TimestampedTranscriptCharacter[] timestampedTranscriptCharacters)> StreamTextToSpeechWithTimestampsAsync(TextToSpeechRequest request, Action<(AudioClip audioClip, TimestampedTranscriptCharacter[] timestampedTranscriptCharacters)> partialClipCallback, CancellationToken cancellationToken = default)
+        {
+            var frequency = request.OutputFormat switch
```

Comment:
Seems like a lot of duplicate code, maybe we can organize this a bit more to reduce dublicates?